crypto/x509/t_x509.c

```diff
@@ -94,11 +94,11 @@ int X509_print_ex(BIO *bp, X509 *x, unsigned long nmflags,
         }
         if (l != -1) {
             if (bs->type == V_ASN1_NEG_INTEGER) {
-                l = -l;
+                l = -(unsigned long)l;
```

Comment:
Casting to unsigned is really done mod 2^n. So on a 1 complement system (unsigned int)-1will also be 2^32-1 while you might expect 2^32-2 if it just reinterprets the bits.